src/shogun/kernel/AUCKernel.h

```diff
@@ -54,66 +52,20 @@ namespace shogun
 		 */
 		bool setup_auc_maximization();
 
-		/** initialize kernel
-		 *
-		 * @param l features of left-hand side
-		 * @param r features of right-hand side
-		 * @return if initializing was successful
-		 */
-		virtual bool init(std::shared_ptr<Features> l, std::shared_ptr<Features> r);
-
-		/** return what type of kernel we are
-		 *
-		 * @return kernel type AUC
-		 */
-		virtual EKernelType get_kernel_type()
-		{
-			return K_AUC;
-		}
-
-		/** return the kernel's name
-		 *
-		 * @return name AUC
-		 */
-		virtual const char* get_name() const
-		{
-			return "AUCKernel";
-		}
+		inline virtual EKernelType get_kernel_type() { return K_AUC; }
 
-		/** return feature class the kernel can deal with
-		 *
-		 * @return feature class SIMPLE
-		 */
-		virtual EFeatureClass get_feature_class()
-		{
-			return C_DENSE;
-		}
+		inline virtual const char* get_name() const { return "AUCKernel"; }
```

Comment:
`inline` can be removed, it does nothing for us.
if you want to force use `SG_FORCE_INLINE` but I dont think that is needed here

Reply:
Yea, the compiler will do that for you since it’s such a small cost to inline it